src/main/java/appeng/me/storage/BasicCellInventory.java

```diff
@@ -80,27 +103,27 @@ private static boolean isStorageCell( final ItemStack i )
 		return false;
 	}
 
-	public static boolean isCell( final ItemStack i )
+	private boolean isStorageCell( final T input )
 	{
-		if( i == null )
+		if( !input.isItem() )
 		{
 			return false;
 		}
+		return isCell( ( (IAEItemStack) input ).getDefinition() );
+	}
 
-		final Item type = i.getItem();
-		if( type instanceof IStorageCell )
+	@SuppressWarnings( { "rawtypes", "unchecked" } )
+	private static boolean isCellEmpty( ICellInventory inv )
+	{
+		if( inv != null )
 		{
-			if( ( (IStorageCell) type ).getChannel() == AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class ) )
-			{
-				return ( (IStorageCell) type ).isStorageCell( i );
-			}
+			return inv.getAvailableItems( inv.getChannel().createList() ).isEmpty();
 		}
-
-		return false;
+		return true;
 	}
 
 	@Override
-	public IAEItemStack injectItems( final IAEItemStack input, final Actionable mode, final IActionSource src )
+	public T injectItems( T input, Actionable mode, IActionSource src )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') The Cyclomatic Complexity of this method "injectItems" is 16 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)